lib/starknet_explorer/cache.ex

```diff
@@ -0,0 +1,44 @@
+defmodule StarknetExplorer.IndexCache do
+  use Agent
+
+  def start_link(_) do
+    blocks_mainnet = StarknetExplorer.Data.many_blocks("mainnet")
+    blocks_testnet = StarknetExplorer.Data.many_blocks("testnet")
+    blocks_testnet2 = StarknetExplorer.Data.many_blocks("testnet2")
+
+    Agent.start_link(
+      fn ->
+        %{
+          "mainnet" => blocks_mainnet,
+          "testnet" => blocks_testnet,
+          "testnet2" => blocks_testnet2
+        }
+      end,
+      name: __MODULE__
+    )
+  end
+
+  def latest_blocks(network) do
+    Agent.get(__MODULE__, fn state -> state[Atom.to_string(network)] end)
+  end
+
+  def add_block(block_number, network) do
+    Agent.update(__MODULE__, fn state ->
+      ## We fetch from the database here just to make the map keys atoms instead of strings
+      ## Yes, really.
```

Comment:
Maybe I'm missing something but can we not use `StarknetExplorerWeb.Utils.atomize_keys()` since it's already handily available?

Reply:
Discussed offline, `rename_rpc_fields |> atomize_keys` might solve it but left for future merges